ee/localserver/krypto-ec-middleware.go

```diff
@@ -113,20 +119,28 @@ func (e *kryptoEcMiddleware) sendCallback(req *http.Request, data *callbackDataS
 
 	resp, err := client.Do(req)
 	if err != nil {
-		level.Debug(e.logger).Log("msg", "got error in callback", "err", err)
+		e.slogger.Log(req.Context(), slog.LevelError,
+			"got error in callback",
+			"err", err,
+		)
 		return
 	}
 
 	if resp != nil && resp.Body != nil {
 		resp.Body.Close()
 	}
 
-	level.Debug(e.logger).Log("msg", "Finished callback", "response-status", resp.Status)
+	e.slogger.Log(req.Context(), slog.LevelDebug, "finished callback",
+		"response-status", resp.Status,
+	)
 }
 
 func (e *kryptoEcMiddleware) Wrap(next http.Handler) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		ctx, span := traces.StartSpan(r.Context())
+
+		spanCtx, span := traces.StartSpan(r.Context())
+		r = r.WithContext(context.WithValue(spanCtx, multislogger.SpanIdKey, span.SpanContext().SpanID().String()))
```

Comment:
wonder if there is a way to make multislogger middleware just inspect the context for the otel span info and pull it out instead of having to manually set it when ever we start a trace

Reply:
figured it out!